File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/candidate/CandidatePersistenceAdapter.java
package kyonggiyo.persistence.candidate;

import kyonggiyo.application.port.out.candidate.DeleteCandidatePort;
import kyonggiyo.application.port.out.candidate.LoadCandidatePort;
import kyonggiyo.application.port.out.candidate.SaveCandidatePort;
import kyonggiyo.domain.candidate.Candidate;
import kyonggiyo.domain.candidate.Status;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class CandidatePersistenceAdapter implements LoadCandidatePort, SaveCandidatePort, DeleteCandidatePort {

    private final CandidateRepository repository;

    @Override
    public Candidate getById(Long id) {
        return repository.getById(id);
    }

    @Override
    public Slice<Candidate> findAllByStatus(Status status, Pageable pageable) {
        return repository.findAllByStatus(status, pageable);
    }

    @Override
    public Candidate save(Candidate candidate) {
        return repository.save(candidate);
    }

    @Override
    public void deleteById(Long id) {
        repository.deleteById(id);
    }

}

File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/candidate/adapter/CandidatePersistenceAdapter.kt
package kyonggiyo.candidate.adapter

import kyonggiyo.application.candidate.domain.entity.Candidate
import kyonggiyo.application.candidate.domain.vo.Status
import kyonggiyo.application.candidate.port.outbound.DeleteCandidatePort
import kyonggiyo.application.candidate.port.outbound.LoadCandidatePort
import kyonggiyo.application.candidate.port.outbound.SaveCandidatePort
import kyonggiyo.candidate.repository.CandidateRepository
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Slice
import org.springframework.stereotype.Component

@Component
class CandidatePersistenceAdapter(
        private val repository: CandidateRepository
) : LoadCandidatePort, SaveCandidatePort, DeleteCandidatePort {

    override fun getById(id: Long): Candidate {
        return repository.getById(id)
    }

    override fun findAllByStatus(status: Status, pageable: Pageable): Slice<Candidate> {
        return repository.findAllByStatus(status, pageable)
    }

    override fun save(candidate: Candidate): Candidate {
        return repository.save(candidate)
    }

    override fun deleteById(id: Long) {
        repository.deleteById(id)
    }

}
